MPI/isx.c

```diff
@@ -84,21 +85,30 @@ int main(int argc,  char ** argv)
 // to set all necessary runtime values and options
 static char * parse_params(const int argc, char ** argv)
 {
-  if(argc != 3)
-  {
-    if( my_rank == 0){
-      printf("Usage:  \n");
-      printf("  ./%s <total num keys(strong) | keys per pe(weak)> <log_file>\n",argv[0]);
-    }
-    exit(1);
-  }
-
   NUM_PES = (uint64_t) comm_size;
   MAX_KEY_VAL = DEFAULT_MAX_KEY;
   NUM_BUCKETS = NUM_PES;
   BUCKET_WIDTH = (uint64_t) ceil((double)MAX_KEY_VAL/NUM_BUCKETS);
-  char * log_file = argv[2];
   char scaling_msg[64];
+  char * log_file;
+
+  if(argc == 3) {
+    NUM_ITERATIONS = 1u;
+    log_file = argv[2];
+  }
+  else if(argc == 4) {
+    NUM_ITERATIONS = (uint64_t) strtoull(argv[2], NULL, 10);
+    log_file = argv[3];
+  }
+
```

Comment:
Thanks @jdinan and @kholland-intel. LGTM. 
@kholland-intel One minor thing. Could you please remove this extra line?